src/core/search/base.cc

```diff
@@ -37,4 +39,11 @@ WrappedStrPtr::operator std::string_view() const {
   return std::string_view{ptr.get(), std::strlen(ptr.get())};
 }
 
+std::optional<double> ParseNumericField(std::string_view value) {
+  double value_as_double;
+  if (absl::SimpleAtod(value, &value_as_double) || value.empty())
```

Comment:
` || value.empty()` looks like a bug

Reply:
Why?

From Redis:
```
127.0.0.1:6379> json.set j1 . '{"value":1}'
OK
127.0.0.1:6379> json.set j2 . '{"another_value":"not number"}'
OK
127.0.0.1:6379> ft.create index on json schema $.value as value numeric
OK
127.0.0.1:6379> ft.search index "*"
1) (integer) 2
2) "j2"
3) 1) "$"
   2) "{\"another_value\":\"not number\"}"
4) "j1"
5) 1) "$"
   2) "{\"value\":1}"
```

So here Redis indexes `j2`, even if it doesn't have the required field (i.e., the field value is an empty string).

Reply:
Yes, I found the edge case: `json.set j1 . '{"value":""}'`